services/blockchain-indexer/shared/dataService/business/schemas.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { requestConnector } = require('../../utils/request');
+
+let allSchemas;
+
+const getSchemasFromNode = async () => {
+	const systemMetadata = await requestConnector('getSystemMetadata');
+	const schemas = await requestConnector('getSchema');
+	return { schemas, systemMetadata };
+};
+
+const getAllSchemas = async () => {
+	if (!allSchemas) {
+		const schemas = await getSchemasFromNode();
+
+		const commandsParamsSchemas = [];
+		const assetsSchemas = [];
+		const eventsSchemas = [];
+
+		schemas.systemMetadata.modules.forEach(module => {
```

Comment:
Could it be converted to a bluebird promise?

Reply:
The usage of promise here would be unnecessary.